resources/ansible/roles/metrics/tasks/main.yml

```diff
@@ -0,0 +1,83 @@
+---
+# The role is making the assumption that telegraf is installed in the base machine image.
+# This is to avoid locking errors with apt.
+
+- name: ensure target directory exists
+  file:
+    path: "{{ network_dashboard_repo_path }}"
+    state: directory
+
+- name: write private key to file
+  copy:
+    content: "{{ network_dashboard_private_key }}"
+    dest: "{{ network_dashboard_sk_path }}"
+    mode: '0600'
+
+- name: check if network dashboard repo already cloned
+  stat:
+    path: "{{ network_dashboard_repo_path }}/.git"
+  register: repo_stat
+
+- name: clone the network dashboard repo
+  git:
+    repo: "{{ network_dashboard_github_url }}"
+    dest: "{{ network_dashboard_repo_path }}"
+    version: main
+    key_file: "{{ network_dashboard_sk_path }}"
+    accept_hostkey: yes
+  when: not repo_stat.stat.exists
+
+- name: ensure _telegraf can run safenode-manager without a password
+  lineinfile:
+    path: /etc/sudoers
+    state: present
+    line: "_telegraf ALL=NOPASSWD: /usr/local/bin/safenode-manager"
+    validate: "visudo -cf %s"
+
+- name: replace SAFENODE_TESTNET_NAME value
+  replace:
+    path: "{{ network_dashboard_repo_path }}/telegraf/EnvironmentFile/telegraf"
+    regexp: 'SAFENODE_TESTNET_NAME=UNDEFINED'
+    replace: 'SAFENODE_TESTNET_NAME={{ testnet_name }}'
+
+- name: replace SAFENODE_BRANCH_NAME value
+  replace:
+    path: "{{ network_dashboard_repo_path }}/telegraf/EnvironmentFile/telegraf"
+    regexp: "SAFENODE_BRANCH_NAME=UNDEFINED"
+    replace: "SAFENODE_BRANCH_NAME=stable"
+
+- name: replace SAFENODE_BRANCH_COMMIT value
+  replace:
+    path: "{{ network_dashboard_repo_path }}/telegraf/EnvironmentFile/telegraf"
+    regexp: "SAFENODE_BRANCH_COMMIT=UNDEFINED"
+    replace: "SAFENODE_BRANCH_COMMIT=c9a381ef7bba8bf9255ae1df006afb1ec33f8464"
```

Comment:
In prior manual entries here against manual internal dashboards, its been just the the first 7 hex characters of the commit.

Is there some shorthand we can stick to here (otherwise, it would then show up as quite wide in the drop down on the dashboard if providing the full commit id as a hex string)?

Reply:
As discussed, this is being left as the undefined value for the moment. We will make some extensions later so that we can obtain the branch and commit hash from the `safenode` binary.